pkg/controller/controller_utils.go

```diff
@@ -1073,19 +1073,19 @@ func AddOrUpdateTaintOnNode(ctx context.Context, c clientset.Interface, nodeName
 	if len(taints) == 0 {
 		return nil
 	}
-	firstTry := true
+
 	return clientretry.RetryOnConflict(UpdateTaintBackoff, func() error {
 		var err error
 		var oldNode *v1.Node
-		// First we try getting node from the API server cache, as it's cheaper. If it fails
-		// we get it from etcd to be sure to have fresh data.
-		option := metav1.GetOptions{}
-		if firstTry {
-			option.ResourceVersion = "0"
-			firstTry = false
-		}
-		oldNode, err = c.CoreV1().Nodes().Get(ctx, nodeName, option)
-		if err != nil {
+
+		getErr := clientretry.OnError(UpdateTaintBackoff, isError, func() error {
```

Comment:
this has two nested retries statements now, that seems considerable more complex to understand

Reply:
I have updated.